walnut/cmd/check.go

```diff
@@ -36,26 +37,19 @@ import (
 // checkCmd represents the check command
 var checkCmd = &cobra.Command{
 	Use:   "check",
-	Short: "check the file passed against a specified schema.",
-	Long: `The check command validates the given file against a specified 
-schema. This file can be a task template, a workflow template or an O² 
-DPL Dump. Each of those have a schema provided to validate against. 
+	Short: "validate the file against a specified schema",
+	Long: `The check command validates the given file against a specified schema. This file can be a task template, 
+a workflow template or an O² DPL Dump. Each of these have a schema to validate against.
 
-Usage:
-  walnut check --format [format] [file]
-
-Example:
-  walnut check --format workflow readout-sftb.yaml
-
-Valid schemata formats:
+Valid schema formats:
   workflow  task  dpl_dump`,
 
 	Run: func(cmd *cobra.Command, args []string) {
-		// fmt.Println("check called with arg: " + strings.Join(args, " "))
-
 		format, _ := cmd.Flags().GetString("format")
-		for _, filename := range args {
-			file, err := ioutil.ReadFile(filename)
+		filename, _ := cmd.Flags().GetStringArray("filename")
```

Comment:
The filenames should come from cobra args instead of a `--filename` flag.
The best practice for command line programs is `command [args] [flags]` or `command [flags] [args]`:
* arguments express *on what* we want to run the operation
* flags modify *how* we want the operation to be executed.

Of course this doesn't apply cleanly to every problem, but I think in this case it does and should.